### ISSUE TITLE
MOE aux loss bug

### ISSUE THREAD

**Issue by mst272** [huggingface/trl#4070]
### Reproduction

I saw that trl added the aux loss for moe calculation:
```
        # Add auxiliary loss if available
        if self.aux_loss_enabled and self.aux_loss_coef:
            aux_loss = self.aux_loss_coef * outputs.aux_loss
            loss += aux_loss
```

However, in the implementation of qwen3 moe in the Transformer, it can be seen that the final loss has already added the aux_loss.
https://github.com/huggingface/transformers/blob/31ab7168ff7e07f61c90134e5238c4d97606aa70/src/transformers/models/qwen3_moe/modeling_qwen3_moe.py#L686

So, is the aux loss added repeatedly to the loss in trl?

### System Info

trl v0.23.0

### Checklist

- [x] I have checked that my issue isn't already filed (see [open issues](https://github.com/huggingface/trl/issues?q=is%3Aissue))
- [x] I have included my system information
- [x] Any code provided is minimal, complete, and reproducible ([more on MREs](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any code provided is properly formatted in code blocks, (no screenshot, [more on code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any traceback provided is complete

**Comment by pramodith** [comment]
That's a good catch! I see the same for GPT-OSS [too](https://github.com/huggingface/transformers/blob/31ab7168ff7e07f61c90134e5238c4d97606aa70/src/transformers/models/gpt_oss/modeling_gpt_oss.py), GraniteMoE (https://github.com/huggingface/transformers/blob/31ab7168ff7e07f61c90134e5238c4d97606aa70/src/transformers/models/granitemoe/modeling_granitemoe.py), PhiMoE (https://github.com/huggingface/transformers/blob/31ab7168ff7e07f61c90134e5238c4d97606aa70/src/transformers/models/phimoe/modeling_phimoe.py), I'm guessing most MoE models have the `aux_loss` included in their final loss.

I can work on reverting that change.
